### PR TITLE
Hugging face pipeline

### LINKED ISSUE #1945: Tutorial for hugging face pipeline.
**Is your feature request related to a problem? Please describe.**
[Hugging Face](https://huggingface.co/) is a machine learning and data science platform and community that helps users build, deploy and train machine learning models. Users can browse through models and data sets that other people have uploaded. Hugging Face is often called the GitHub of machine learning because it lets developers share and test their work openly. Currently there are more than 1 million models shared on hugging face.

It would be more convenient if MONAI also has a tutorial to show how to wrap a MONAI task like segmentation into a hugging face pipeline. Then user can:
1. Conveniently use the MONAI task as a hugging face pipeline.
2. Distribute their MONAI based models into hugging face so that more users can get access. 

**Describe the solution you'd like**
A tutorial to show how to use MONAI to write a hugging face pipeline.

### PR DESCRIPTION
**Pull request by binliunls**
Fixes #1945 .

### Description
Add a tutorial to show how to rewrite a spleen ct segmentation task into a hugging face pipeline so that users can easily call and distribute it.

### Checks

- [ ] Avoid including large-size files in the PR.
- [ ] Clean up long text outputs from code cells in the notebook.
- [ ] For security purposes, please check the contents and remove any sensitive info such as user names and private key.
- [ ] Ensure (1) hyperlinks and markdown anchors are working (2) use relative paths for tutorial repo files (3) put figure and graphs in the `./figure` folder
- [ ] Notebook runs automatically `./runner.sh -t <path to .ipynb file>`

### CODE REVIEW

**Comment by Nic-Ma** [comment]
As discussed offline, @binliunls will add more charts.

Thanks.

**Comment by binliunls** [comment]
> As discussed offline, @binliunls will add more charts.
> 
> Thanks.

Updated

File: hugging_face/finetune_vista3d_for_hugging_face_pipeline.ipynb [hunk 1]
```diff
@@ -0,0 +1,426 @@
+{
```
Comment (KumoLiu) [inline hugging_face/finetune_vista3d_for_hugging_face_pipeline.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #7.        numpy_output = np.squeeze(numpy_output, axis=0)
</pre><p>Just FYI we can use <code>SqueezeDim</code> in the MONAI here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/1946/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (KumoLiu) [inline hugging_face/finetune_vista3d_for_hugging_face_pipeline.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #10.    vista3d_hf_model.network.load_state_dict(torch.load(fine_tuned_model_path))
</pre><p> </p><p>Would be better to add <code>weights_only=True</code> in the <code>torch.load</code> to avoid the error after pytorch 2.6.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/1946/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

File: hugging_face/hugging_face_pipeline_for_monai.ipynb [hunk 2]
```diff
@@ -0,0 +1,584 @@
+{
```
Comment (KumoLiu) [inline hugging_face/hugging_face_pipeline_for_monai.ipynb:1]: <p>Missing remove tempdir in this notebook</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/1946/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (KumoLiu) [inline hugging_face/hugging_face_pipeline_for_monai.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #8.    monai_unet.unet.load_state_dict(torch.load(weight_dir))
</pre><p>The same as before.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/1946/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

**Review by Nic-Ma — APPROVED** [review]
Thanks for the quick update.
Looks good to me now.

File: hugging_face/finetune_vista3d_for_hugging_face_pipeline.ipynb (hunk 1 above)
Comment (binliunls) [reply hugging_face/finetune_vista3d_for_hugging_face_pipeline.ipynb:1]: Thanks for this head-up. This function here is to transter the output to numpy for display purpose. I think the numpy squeeze should be fine.
